core/computing-unit-managing-service/src/main/scala/edu/uci/ics/texera/service/resource/ComputingUnitAccessResource.scala

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package edu.uci.ics.texera.service.resource
+
+import edu.uci.ics.texera.auth.SessionUser
+import edu.uci.ics.texera.dao.SqlServer
+import edu.uci.ics.texera.dao.SqlServer.withTransaction
+import edu.uci.ics.texera.dao.jooq.generated.enums.{PrivilegeEnum, WorkflowComputingUnitTypeEnum}
+import edu.uci.ics.texera.dao.jooq.generated.tables.daos.{
+  ComputingUnitUserAccessDao,
+  UserDao,
+  WorkflowComputingUnitDao
+}
+import edu.uci.ics.texera.dao.jooq.generated.tables.pojos.ComputingUnitUserAccess
+import edu.uci.ics.texera.service.resource.ComputingUnitManagingResource.{
+  DashboardWorkflowComputingUnit,
+  context
+}
+import edu.uci.ics.texera.service.util.KubernetesClient
+import edu.uci.ics.texera.service.resource.ComputingUnitAccessResource._
+import edu.uci.ics.texera.service.util.ComputingUnitHelpers.{
+  getComputingUnitMetrics,
+  getComputingUnitStatus
+}
+import edu.uci.ics.texera.dao.jooq.generated.Tables.COMPUTING_UNIT_USER_ACCESS
+
+import scala.jdk.CollectionConverters._
+import io.dropwizard.auth.Auth
+import jakarta.annotation.security.RolesAllowed
+import jakarta.ws.rs.core.{MediaType, Response}
+import jakarta.ws.rs.{GET, PUT, Path, PathParam, Produces, DELETE}
+import org.jooq.{DSLContext, EnumType}
+
+object ComputingUnitAccessResource {
+  private lazy val context: DSLContext = SqlServer
```

Comment:
All the endpoints here should check if the `share-computing-unit-enabled` is true or false. If it is false, reject all the incoming request